### PR TITLE
Module 6

### PR DESCRIPTION
**Pull request by jaybee1971**
All diagrams added to new 'Documentation' folder and created class diagram as part of stretch goal by exploring the plantUML plugin for VSCode (markdown used to create the diagram also in the folder).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,6 +71,11 @@ To get this app to work for you, you will need to have a Trello account and stor
   * API_TOKEN  -  Generate your own API token and store value here
   * BOARD_ID  -  The ID of the board you want to use for the project
 ```
+For this application:
+```bash
+my_statuses = ["Not Started", "In Progress", "Completed"]
+```
```
Comment (Jonesey13) [inline README.md:77]: Good job for documenting this btw!

File: Documentation/class_diagram.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+### uml: class diagram
```
Comment (Jonesey13) [inline Documentation/class_diagram.md:1]: If you wanted to go further you could cover your other classes.

**Review by Jonesey13 — CHANGES_REQUESTED** [review]
Main issue blocking this from passing is the component diagram (which doesn't cover the structure of your code). More specific comments below:

**Context Diagram**
This all looks very good

**Container Diagram**
The webpage in the user's browser (which is arguably not an single page app (SPA)) must go through the gunicorn server to communicate with the todo app backend (not really an API either)

**Component Diagram**
This refers to API's way too much when they simply aren't present (the only valid use on the diagram is arguably the Trello API. This diagram should reflect the high level structure of your code (so it can include routes as one block, but don't split it into methods e.g. GET, POST etc.). What you should include in categories like templates, models, view models, the trello api module and so on)

**Class Diagram**
What you have is fine but could be extended to go beyond your models to include the other python files.

**Comment by jaybee1971** [comment]
@Jonesey13 I have pushed v2 versions of the container and component diagram to the repo, please review and feedback.

File: README.md (hunk 1 above)
Comment (jaybee1971) [reply README.md:77]: I have pushed v2 versions of the container and component diagram to the repo, please review and feedback.

**Review by jaybee1971 — COMMENTED** [review]
I have created v2 versions of the diagrams for review

**Review by JackMeadDev — APPROVED** [review]
These updated diagrams show a better understanding of the code structure and I'm happy for those to be merged, but I think there's still a few details worth considering:

Container:
* There's still the suggestion that the flask app provides the web page (via the web environment) without being accessed through gunicorn - actually the application itself is building the templates and providing them back to the user, as well as handling the core business logic (so all requests go through gunicorn, then flask, then to our app including initial requests for the web page)

Component
* The web page doesn't build the templates using the view model; the application (via app.py) does that and then returns them to the user as the web page, and the user is then more easily able to access other routes through app.py
* It might be clearer to cut the amount of the information in the boundary that you have, e.g. show incoming requests at the component level arriving from gunicorn or flask rather than from a web page directly since this hides the gunicorn/flask boundary a bit, and to the application it's not actually important where those requests arrived from outside of that

**Pull request merged**